libs/types/src/tokens.rs

```diff
@@ -204,6 +204,48 @@ pub struct CustomMetadata {
 
 	/// Whether an asset can be used as a currency to fund Centrifuge Pools.
 	pub pool_currency: bool,
+
+	/// The ways this token is cross-chain transferable
+	pub transferability: CrossChainTransferability,
+}
+
+#[derive(
+	Clone,
+	Copy,
+	Default,
+	PartialOrd,
+	Ord,
+	PartialEq,
+	Eq,
+	Debug,
+	Encode,
+	Decode,
+	TypeInfo,
+	MaxEncodedLen,
+)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub enum CrossChainTransferability {
+	/// The asset is only transferable through XCM
+	Xcm,
+
+	/// The asset is only transferable through Centrifuge Connectors
+	Connectors,
+
+	/// The asset is transferable through all available options.
+	/// NOTE: Once set to `All`, the asset is automatically transferrable
+	/// through any eventual new option being added at a later stage. A
+	/// migration might be required if that's undesirable.
+	All,
+
+	/// The asset is not transferable cross-chain
+	#[default]
+	None,
+}
+
+impl CrossChainTransferability {
+	pub fn includes(self, other: Self) -> bool {
+		self == other || self == Self::All
```

Comment:
We must add test(s) as the current check is also positive when checking `CrossChainTransferability::All.includes(CrossChainTransferability::None)`.

Reply:
What is potentially wrong with that expression being `true`? I can't think of any use case that could turn out dangerous or erroneous 🤔

Reply:
Oh, I see. `All` should not imply `None`, so this indeed undesired from a logical perspective. Will tackle!

Reply:
@wischli I've made `transferability` option: https://github.com/centrifuge/centrifuge-chain/pull/1357/commits/2dc11a9b9585d776016ddcf1aeb4fc005ce4be26